Userland/Libraries/LibC/unistd.cpp

```diff
@@ -398,11 +402,72 @@ ssize_t pwrite(int fd, const void* buf, size_t count, off_t offset)
     return nwritten;
 }
 
+// Note: Be sure to send to directory_name parameter a directory name ended with trailing slash.
+static int ttyname_r_for_directory(const char* directory_name, dev_t device_mode, ino_t inode_number, char* buffer, size_t size)
+{
+    DIR* dirstream = opendir(directory_name);
+    if (!dirstream) {
+        return -1;
+    }
+
+    auto close_dir_stream_on_exit = ScopeGuard([dirstream] {
+        closedir(dirstream);
+    });
+
+    struct stat st;
+    struct dirent* entry;
+    char* name_path;
```

Comment:
Please declare variables as late as possible. We are not bound by old C rules here :)
Also please initialize primitive variables explicitly instead of leaving them uninitialized.